testutil-server/src/main/java/io/spine/testing/server/blackbox/Dashboard.java

```diff
@@ -0,0 +1,144 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.testing.server.blackbox;
+
+import com.google.common.flogger.FluentLogger;
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+import io.spine.base.EventMessage;
+import io.spine.base.Identifier;
+import io.spine.core.MessageId;
+import io.spine.core.Subscribe;
+import io.spine.logging.Logging;
+import io.spine.server.event.AbstractEventSubscriber;
+import io.spine.system.server.AggregateHistoryCorrupted;
+import io.spine.system.server.CannotDispatchDuplicateCommand;
+import io.spine.system.server.CannotDispatchDuplicateEvent;
+import io.spine.system.server.ConstraintViolated;
+import io.spine.system.server.HandlerFailedUnexpectedly;
+import io.spine.system.server.RoutingFailed;
+
+import java.io.PrintStream;
+
+import static io.spine.json.Json.toJson;
+import static java.lang.String.format;
+
+/**
+ * A subscriber for all the diagnostic events.
+ *
+ * <p>Logs all the received diagnostic events with a meaningful message.
+ */
+final class Dashboard
+        extends AbstractEventSubscriber
+        implements Logging {
+
+    private static final Dashboard instance = new Dashboard();
+
+    /**
+     * Obtains the only instance of {@code Dashboard}.
+     */
+    static Dashboard instance() {
+        return instance;
+    }
+
+    /**
+     * Prevents direct instantiation.
+     */
+    private Dashboard() {
+        super();
+    }
+
+    @Subscribe
+    void on(ConstraintViolated event) {
+        String typeUrl = event.getEntity()
+                              .getTypeUrl();
+        String idAsString = Identifier.toString(event.getEntity().getId());
+        log(event, "Entity state (ID: %s, type: %s) is invalid.", idAsString, typeUrl);
+    }
+
+    @Subscribe
+    void on(CannotDispatchDuplicateCommand event) {
+        MessageId command = event.getDuplicateCommand();
+        log(event, "Command %s (ID: %s) should not be dispatched twice.",
+            command.getTypeUrl(),
+            command.asCommandId()
+                   .getUuid());
+    }
+
+    @Subscribe
+    void on(CannotDispatchDuplicateEvent event) {
+        MessageId duplicateEvent = event.getDuplicateEvent();
+        log(event, "Event %s (ID: %s) should not be dispatched twice.",
+            duplicateEvent.getTypeUrl(),
+            duplicateEvent.asEventId()
+                          .getValue());
+    }
+
+    @Subscribe
+    void on(HandlerFailedUnexpectedly event) {
+        log(event, "Signal %s could not be handled by %s:%n%s",
```

Comment:
As we're putting messages on new line (and understandably not having a period in the format) we need to separate this message from the next one on the log.

Options that I see here:
 1. Have some indentation after `%n`.
 2. Have another `%n` at the end, so that the whole log record on an event looks as a clearly separated block.

Without it the message which starts at the new line could be confused with a start of a new log record. It's easy to miss the colon at the end of the previous line, especially if the log line is long and full of code-related details.